haystack/management/commands/build_solr_schema.py

```diff
@@ -2,57 +2,140 @@
+            help='If provided, attempts to configure a core located in the \
+            given directory by removing the managed-schema.xml(renaming) if it \
+            exists, configuring the core by rendering the schema.xml and \
+            solrconfig.xml templates provided in the django project\'s \
+            TEMPLATE_DIR/search_configuration DIR\'s'
+        )
+        parser.add_argument(
+            "-r", "--reload-core",
+            help='If provided, attempts to automatically reload the solr core \
+            via the urls in the \'URL\' and \'ADMIN_URL\' settings of the SOLR \
+            HAYSTACK_CONNECTIONS entry. BOTH MUST be provided'
+        )
 
     def handle(self, **options):
         """Generates a Solr schema that reflects the indexes."""
         using = options.get('using')
-        schema_xml = self.build_template(using=using)
+        if not isinstance(connections[using].get_backend(), SolrSearchBackend):
+            raise ImproperlyConfigured("'%s' isn't configured as a SolrEngine)." % connections[using].get_backend().connection_alias)
+
+        schema_xml = self.build_template(using=using, template_filename=Command.schema_template_loc)
+        solrcfg_xml = self.build_template(using=using, template_filename=Command.solrcfg_template_loc)
+
+        filename = options.get('filename')
+        configure_directory = options.get('configure_directory')
+        reload_core = options.get('reload_core')
+
+        if filename:
+            self.stdout.write("Trying to write schema file located at {}".format(filename))
+            self.write_file(filename, schema_xml)
+
+            if reload_core:
+                connections[using].get_backend().reload()
+
+        if configure_directory:
+            self.stdout.write("Trying to configure core located at {}".format(configure_directory))
+
+            managed_schema_path = os.path.join(configure_directory, 'managed-schema')
+
+            if os.path.isfile(managed_schema_path):
+                try:
+                    os.rename(managed_schema_path, '%s.old' % managed_schema_path)
+                except:
+                    raise CommandError('Could not rename old managed schema file out of the way: {}'.format(managed_schema_path))
+
+            schema_xml_path = os.path.join(configure_directory, 'schema.xml')
+
+            try:
+                self.write_file(schema_xml_path, schema_xml)
+            except EnvironmentError as exc:
+                raise CommandError('Could not configure {}: {}'.format(schema_xml_path, exc))
+
+            solrconfig_path = os.path.join(configure_directory, 'solrconfig.xml')
+
+            try:
+                self.write_file(solrconfig_path, solrcfg_xml)
+            except EnvironmentError as exc:
+                raise CommandError('Could not write {}: {}'.format(solrconfig_path, exc))
+
+        if reload_core:
+            core = settings.HAYSTACK_CONNECTIONS['solr']['URL'].rsplit('/', 1)[-1]
```

Comment:
This line throws an exception if the core is labeled as `'default'` instead of `'solr'` in `HAYSTACK_CONNECTIONS`. Is this the desired behavior?